### PR TITLE
Grab DC if within a folder

### PR DESCRIPTION
**Pull request by chris1984**

### CODE REVIEW

File: lib/fog/vsphere/models/compute/datacenter.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@ def storage_pods(filters = {})
         end
 
         def vm_folders(filters = {})
-          service.folders({ datacenter: path.join('/'), type: :vm }.merge(filters))
+          service.folders({ datacenter: name, type: :vm }.merge(filters))
```
Comment (ezr-ondrej) [inline lib/fog/vsphere/models/compute/datacenter.rb]: Could we do it in all the methods, please? I guess we have the same bug in all of them.

**Comment by chris1984** [comment]
@ezr-ondrej updated :)

**Comment by chris1984** [comment]
fixing rubocop

**Comment by ezr-ondrej** [comment]
As discussed offline, this doesn't work in all cases, we should make sure we are covering all the use cases first.

**Comment by chris1984** [comment]
Getting this from Foreman:

http://rx-paste.usersys.redhat.com/view/1748dc7c

**Comment by chris1984** [comment]
@ezr-ondrej can we merge this for now then work on the real issue later to close the bugzilla?

Tested it with the original code submit and it works fine:

Foreman host created:

https://nimb.ws/txXu1s

Host created in vCenter:

https://nimb.ws/eOmdT7

**Comment by ezr-ondrej** [comment]
Ok, as this is fixing the issue in the folder and it doesn't break anything... The issues i had having are unrelated it seems and they appear even without the patch.

**Pull request merged**